wicket-core/src/main/java/org/apache/wicket/csp/CSPRequestCycleListener.java

```diff
@@ -39,14 +40,7 @@ public CSPRequestCycleListener(ContentSecurityPolicySettings settings)
 	}
 
 	@Override
-	public void onRequestHandlerResolved(RequestCycle cycle, IRequestHandler handler)
-	{
-		// WICKET-7028- this is needed for redirect to buffer use case.
-		protect(cycle, handler);
-	}
-
-	@Override
-	public void onRequestHandlerExecuted(RequestCycle cycle, IRequestHandler handler)
+	public void onUrlMapped(RequestCycle cycle, IRequestHandler handler, Url url)
```

Comment:
I have tried this with 

https://issues.apache.org/jira/browse/WICKET-7028

sample application and this change seems to be Ok. But to be honest I'm not sure this is the way we should go.  See 

https://issues.apache.org/jira/browse/WICKET-7040

Reply:
WICKET-7040 is never going to work, I think. Due to ajax requests, components can be instantiated long after the original page has been created and long after the original CSP has been sent to the browser. Additional CPSs could be delivered to the client as meta tags but weakening the original CSP is not allowed ( https://www.w3.org/TR/CSP3/#multiple-policies ). Therefore the strictest possible set of policies has to be known before the page is rendered.